docs/upgrade.html

```diff
@@ -25,6 +25,10 @@ <h5><a id="upgrade_320_notable" href="#upgrade_320_notable">Notable changes in 3
             which meant that idempotence remained disabled unless the user had explicitly set <code>enable.idempotence</code> to true
             (See <a href="https://issues.apache.org/jira/browse/KAFKA-13598">KAFKA-13598</a>for more details).
             This issue was fixed and the default is properly applied in 3.0.1, 3.1.1, and 3.2.0.</li>
+        <li>A notable exception is Connect that by default currently disables idempotence for its
+            producers in order to support connecting to older versions of the Kafka broker without
+            explicit changes. Enabling idempotence is still possible via configuration and this
+            default might change in a major version upgrade in the future.</li>
```

Comment:
I also think it's worth clarifying the same thing here. Maybe:
```suggestion
        <li>A notable exception is Connect that by default disables idempotent behavior for all of its
            producers in order to uniformly support using a wide range of Kafka broker versions. 
            Users can change this behavior to enable idempotence for some or all producers
            via Connect worker and/or connector configuration. Connect may enable idempotent producers
            by default in a future major release.</li>
```
WDYT?